ocs_ci/deployment/baremetal.py

```diff
@@ -22,7 +36,471 @@ class BAREMETALUPI(Deployment):
+            image_data = rhcos_images_file[ocp_version]
+            # Download installer_initramfs
+            initramfs_image_path = constants.coreos_url_prefix + image_data['installer_initramfs_url']
+            if check_for_rhcos_images(initramfs_image_path):
+                cmd = (
+                    "wget -O "
+                    f"{self.helper_node_details['bm_tftp_dir']}"
+                    "/rhcos-installer-initramfs.x86_64.img "
+                    f"{initramfs_image_path}"
+                )
+                assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to Download required File"
+            else:
+                raise RhcosImageNotFound
+            # Download installer_kernel
+            kernel_image_path = constants.coreos_url_prefix + image_data['installer_kernel_url']
+            if check_for_rhcos_images(kernel_image_path):
+                cmd = (
+                    "wget -O "
+                    f"{self.helper_node_details['bm_tftp_dir']}"
+                    "/rhcos-installer-kernel-x86_64 "
+                    f"{kernel_image_path}"
+                )
+                assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to Download required File"
+            else:
+                raise RhcosImageNotFound
+            # Download metal_bios
+            metal_image_path = constants.coreos_url_prefix + image_data['metal_bios_url']
+            if check_for_rhcos_images(metal_image_path):
+                cmd = (
+                    "wget -O "
+                    f"{self.helper_node_details['bm_path_to_upload']}"
+                    "/rhcos-metal.x86_64.raw.gz "
+                    f"{metal_image_path}"
+                )
+                assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to Download required File"
+            else:
+                raise RhcosImageNotFound
+            # Create pxelinux.cfg directory
+            cmd = f"mkdir -m 755 {self.helper_node_details['bm_tftp_dir']}/pxelinux.cfg"
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to create required folder"
+
+        def deploy(self, log_cli_level='DEBUG'):
+            """
+            Deploy
+            """
+            # Uploading pxe files
+            logger.info("Deploying OCP cluster for Bare Metal platform")
+            logger.info(
+                f"Openshift-installer will be using log level:{log_cli_level}"
+            )
+            upload_file(
+                self.host,
+                constants.COMMON_CONF_FILE,
+                os.path.join(
+                    self.helper_node_details['bm_dnsmasq_dir'],
+                    "dnsmasq.common.conf"
+                ),
+                self.user,
+                key_file=self.private_key
+            )
+            logger.info("Uploading PXE files")
+            for machine in self.mgmt_details:
+                if self.mgmt_details[machine].get('role') == "bootstrap":
+                    upload_file(
```

Comment:
consider also here the dict approach suggested above

Reply:
I guess here we cannot use the dict approach